python/core/auto_generated/qgsmaplayer.sip.in

```diff
@@ -726,6 +726,74 @@ Read all custom properties from layer. Properties are stored in a map and saved
 .. versionadded:: 3.14
 %End
 
+    virtual int listStylesInDatabase( QStringList &ids /Out/, QStringList &names /Out/,
```

Comment:
These methods shouldn't actually be part of QgsMapLayer (or QgsVectorLayer) at all. (We shouldn't have to construct a layer in order to retrieve or delete styles). Instead they belong in QgsProviderMetadata (there's already listStyles / styleExists / getStyleById / deleteStyleById / saveStyle / loadStyle there, so possible we already have all the API needed).

These methods are currently in QgsVectorLayer for legacy reasons only, not because they belong there. :laughing:

Reply:
So introducing a

    QgsProviderMetadata* QgsMapLayer::providerMetadata() {
        QgsProviderRegistry::instance()->providerMetadata(mProviderKey);
    }

and 

    QgsProviderMetadata* QgsProviderRegistry::providerMetadata( const QString &providerKey ) {
        return findMetadata_( mProviders, providerKey );
    }

and accessing these, where a layer is available, through `layer->providerMetadata()->{listStyles,styleExists,getStyleById,deleteStyleById,saveStyle,loadStyle}`?